crypto/o_time.c

```diff
@@ -105,7 +105,7 @@ struct tm *OPENSSL_gmtime(const time_t *timer, struct tm *result)
 {
     struct tm *ts = NULL;
 
-#if defined(OPENSSL_THREADS) && !defined(OPENSSL_SYS_WIN32) && !defined(OPENSSL_SYS_OS2) && (!defined(OPENSSL_SYS_VMS) || defined(gmtime_r)) && !defined(OPENSSL_SYS_MACOSX) && !defined(OPENSSL_SYS_SUNOS)
+#if defined(OPENSSL_THREADS) && !defined(OPENSSL_SYS_WIN32) && !defined(OPENSSL_SYS_OS2) && (!defined(OPENSSL_SYS_VMS) || defined(gmtime_r)) && !defined(OPENSSL_SYS_SUNOS)
```

Comment:
Oh! Maybe 64-bit VMS is not supported in 1.0.2, in which case one doesn't have go through pointer trouble mentioned above... @levitte, care to weigh in? BTW, `!defined(OPENSSL_SYS_MACOSX)` was contributed as "make OpenSSL compilable on MacOS/X". One can argue "oh! that was long time ago". Yes, but then it better be explicitly said, e.g. in commit message, preferably with reference when they implemented it or at least oldest version one verified. On the other hand one can argue that it's not the kind of things that one should do in branch that is in maintenance mode, i.e. make adjustment based on current status of nearby computer...

Reply:
Revisiting this...  I just realised that under "normal" circumstances, `gmtime_r` is undefined on VMS, so this section never gets built, which means that either the `gmtime` implementation will be used, or the much more VMSy implementation further down.

The condition above is incorrect, `(!defined(OPENSSL_SYS_VMS) || defined(gmtime_r))` should really be `(!defined(OPENSSL_SYS_VMS) || (__CRTL_VER >= 70000000 &&  (defined _POSIX_C_SOURCE || !defined _ANSI_C_SOURCE)))`...  funny thing is that exactly that condition is used to define `VMS_GMTIME_OK`

So I suggest that the VMS stuff needs a bit of renovation which is beyond the scope of this PR.  Or leave it be, let it keep using `gmtime` (which is supposedly thread safe on VMS, due to internal thread local storage).